### DISCUSSION TITLE
Is it possible to add custom system fields or have a flow where fields are added to collections by default?

### DISCUSSION THREAD

**Discussion by MannanK** [directus/directus#14959] [Archive]
Hey all, I had a question regarding system fields & flows. I am using Directus on top of an app that uses Ruby on Rails & PostgreSQL. It has worked great so far, but I just ran into an issue where I created an abstract model in Rails for a table created through the Directus UI, but because of Rails' timestamp fields being called `created_at` & `updated_at` vs. Directus' `date_created` & `date_updated`, I'm running into some issues with some ActiveRecord helpers.

With that in mind, I was wondering if it's possible to add custom system fields so that similar to allowing editors to add optional fields like `status`, `date_created`, etc., we can add additional checkboxes for other predefined fields? If not, would it be possible to introduce some kind of flow where two fields called `created_at` & `updated_at` are added by default whenever a collection is created?

For what it's worth, I did look through the System Collections section, and couldn't find any internal table that housed the info for which fields show up as system fields, unless I'm looking in the wrong place?

Thanks for any help and happy to provide more details if needed!

**Comment by rijkvanzanten** [comment]
The optional system fields are just presets for "Regular" fields you can also configure by hand 🙂 The names are arbitrary as well, we just suggest "date_created" as a starting point, but they can be updated to whatever you want.

There's two things you can do: 

1. Rename the field when creating the collection. The "optional system field" preset inputs aren't just checkboxes, you can actually type in them! The `date_created` name is just a default suggestion
2. Create those fields after the fact. For example `date_created` is nothing but a timestamp type field with the "On Create" schema option set to "Current Date & Time".

There's currently no way however to change what "optional fields" are suggested by default. You might be able to listen to events on the directus_collections collection, and auto-create new fields on the fly in a flow, but I've never tried that myself 🤔

**Reply by MannanK** [reply]
Thanks for the reply @rijkvanzanten! #&#8203;2 would definitely be an option, though I was trying to avoid that as it introduces room for human-error (someone forgets to add the timestamps when creating a new collection, f.e) & puts the onus on the editors to have to remember. Definitely good to know #&#8203;1 as I seem to have missed the fact that those names are editable, but again for similar reasons to #&#8203;2, this would put the onus on the editors to have to remember rather than this being automated.

If there was a way to change the preset text, that would be by far the easiest approach I feel like, but it sounds like this is not possible. I also was not able to figure out a way to use flows to auto-create fields on the fly, but will keep digging on that front.

Let me know if you had any other ideas, but thanks for the help regardless!
